kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -691,6 +699,42 @@ private List<Field> getRegularSchemaFields(com.google.cloud.bigquery.Schema valu
     return result;
   }
 
+  private List<Field> validateAndRestoreFieldTypes(List<Field> proposedFields, FieldList existingFields) {
+    checkNotNull(proposedFields);
+    if (existingFields == null) {
+      return proposedFields;
+    }
+    Set<String> existingFieldNames = existingFields.stream().map(Field::getName).collect(Collectors.toSet());
+    return proposedFields.stream().map(proposedField -> {
+      if (!existingFieldNames.contains(proposedField.getName())) {
+        return proposedField;
+      }
+      Field existingField = existingFields.get(proposedField.getName());
+      if (existingField.getType() == LegacySQLTypeName.RECORD) {
+        if (proposedField.getType() == LegacySQLTypeName.RECORD) {
+          Field.Builder updatedField = existingField.toBuilder();
+          List<Field> updatedSubfields = validateAndRestoreFieldTypes(proposedField.getSubFields(), existingField.getSubFields());
+          updatedField.setType(LegacySQLTypeName.RECORD, FieldList.of(updatedSubfields));
+          return updatedField.build();
+        }
+      } else {
+        if (isCompatibleFieldTypes(proposedField.getType(), existingField.getType())) {
+          return existingField;
+        }
+      }
+      throw new BigQueryConnectException(String.format(
```

Comment:
I am little less confident on failing task here. Can we instead log a warn this and consider `existingField` as source of truth ? 
Any specific reasons for failing apart from fail fast? I understand that any compatible conversions would eventually give us failure from bigquery, but I would prefer to have that behaviour as it is today.